src/iot/commands/createCert.ts

```diff
@@ -0,0 +1,133 @@
+
+    if (certExists) {
+        getLogger().error('Certificate path {0} already exists', certPath)
+        window.showErrorMessage(
+            localize('AWS.iot.createCert.error', 'Failed to create certificate. Path {0} already exists.', certPath)
+        )
+        return undefined
+    }
+    if (privateKeyExists) {
+        getLogger().error('Key path {0} already exists', privateKeyPath)
+        window.showErrorMessage(
+            localize(
+                'AWS.iot.createCert.error',
+                'Failed to create certificate. Path {0} already exists.',
+                privateKeyPath
+            )
+        )
+        return undefined
+    }
+    if (publicKeyExists) {
+        getLogger().error('Key path {0} already exists', publicKeyPath)
+        window.showErrorMessage(
+            localize(
+                'AWS.iot.createCert.error',
+                'Failed to create certificate. Path {0} already exists.',
+                publicKeyPath
+            )
+        )
+        return undefined
+    }
+
+    try {
+        const certificate = await node.iot.createCertificateAndKeys({
+            setAsActive: false,
+        })
+        const certId = certificate.certificateId
+        const certPem = certificate.certificatePem
+        const privateKey = certificate.keyPair?.PrivateKey
+        const publicKey = certificate.keyPair?.PublicKey
+
+        if (!certPem || !privateKey || !publicKey) {
+            getLogger().error('Could not download certificate. Certificate is missing either the PEM or key pair.')
+            showViewLogsMessage(localize('AWS.iot.createCert.error', 'Failed to create certificate'), window)
+            return undefined
+        }
+
+        //Save resources
+        await fs.writeFile(certPath, certPem, { encoding: PEM_FILE_ENCODING, mode: MODE_RW_R_R })
+        await fs.writeFile(privateKeyPath, privateKey, { encoding: PEM_FILE_ENCODING, mode: MODE_RW_R_R })
+        await fs.writeFile(publicKeyPath, publicKey, { encoding: PEM_FILE_ENCODING, mode: MODE_RW_R_R })
+        getLogger().info(`Downloaded certificate ${certId}`)
+    } catch (e) {
+        getLogger().error('Failed to create and save certificate: %O', e)
+        showViewLogsMessage(localize('AWS.iot.createCert.error', 'Failed to create certificate'), window)
+    }
+
+    //Refresh the Certificate Folder node
+    await node.refreshNode(commands)
+}
+
+async function promptForSaveLocation(window: Window): Promise<vscode.Uri | undefined> {
+    // const folderLocation = await window.showOpenDialog({
+    //     openLabel: localize('AWS.iot.downloadCert.openButton', 'Save certificate here'),
+    //     canSelectFolders: true,
+    //     canSelectFiles: false,
+    //     canSelectMany: false,
+    // })
+
+    // if (!folderLocation || folderLocation.length == 0) {
+    //     return undefined
+    // }
+
+    // return folderLocation[0]
+    const saveLocation = await window.showSaveDialog({})
```

Comment:
I used a save dialog here so that the user could choose the base name of the certificate that's saved. However, since we need to save three files and not just one, we append a suffix, e.g `-certificate.pem.crt` or `-private.pem.key`, so the actual saved files don't match exactly what the user gives. E.g. `myCert` becomes `myCert-certificate.pem.crt`, `myCert-private.pem.key`, and `myCert.public.pem.key`.

Alternatively, I thought of using an openDialog and prompting for a folder to save the files, and then saving each of the files as `[64 character hexadecimal ID]-[certificate/private/public].pem.[key/crt]`. Note that when these credentials are downloaded from the AWS console, they use either this format, or formerly, `[10 character truncated ID]-[certificate/private/public].pem.[key/crt]`.

I'm unsure which method would be better, or if there's a more ideal way to prompt the user for this information.

Reply:
Maybe we can prompt for a base name via `createInputBox` that's pre-filled with whatever default name seems reasonable, then prompt for a directory after? Need to let the user know that multiple files are being saved, so we could notify the actual names at some point prior to saving.

Reply:
For now I rewrote this function to act the same way as the AWS Console, i.e. using the full certificate ID. Prompting for the base name via an input box could be an incremental improvement that can be added later